src/roiextractors/extractors/memmapextractors/memmapextractors.py

```diff
@@ -64,43 +62,86 @@ def get_num_channels(self):
         return self._num_channels
 
     @staticmethod
-    def write_imaging(imaging_extractor: ImagingExtractor, save_path: PathType = None, verbose: bool = False):
+    def write_imaging(
+        imaging_extractor: ImagingExtractor,
+        save_path: PathType = None,
+        verbose: bool = False,
+        force_chunk: bool = False,
+    ):
         """
         Static method to write imaging.
 
         Parameters
         ----------
-        imaging: ImagingExtractor object
-            The EXTRACT segmentation object from which an EXTRACT native format
-            file has to be generated.
+        imaging: An ImagingExtractor object that inherited from MemmapImagingExtractor
         save_path: str
-            path to save the native format.
-        overwrite: bool
-            If True and save_path is existing, it is overwritten
+            path to save the native format to.
+        verbose: bool
+            Displays a progress bar.
+        force_chunk: bool
+            Forces chunk to occur even if memmory is available
         """
         imaging = imaging_extractor
-        video_to_save = np.memmap(
-            save_path,
-            shape=(
-                imaging.get_num_frames(),
-                imaging.get_num_channels(),
-                imaging.get_image_size()[0],
-                imaging.get_image_size()[1],
-            ),
-            dtype=imaging.get_dtype(),
-            mode="w+",
-        )
-
-        if verbose:
-            for ch in range(imaging.get_num_channels()):
-                print(f"Saving channel {ch}")
-                for i in tqdm(range(imaging.get_num_frames())):
-                    plane = imaging.get_frames(i, channel=ch)
-                    video_to_save[ch, i] = plane
-        else:
-            for ch in range(imaging.get_num_channels()):
-                for i in range(imaging.get_num_frames()):
-                    plane = imaging.get_frames(i, channel=ch)
-                    video_to_save[ch, i] = plane
+        safety_margin = 0.80  # Accept a file smaller than 80 per cent of available memory
```

Comment:
A more informative name perhaps `memory_safety_margin`?

Reply:
Yes, that's better. Thanks for the suggestion.

Reply:
Done.